x-pack/legacy/plugins/maps/public/reducers/map.js

```diff
@@ -453,10 +453,23 @@ function resetDataRequest(state, action, request) {
     return state;
   }
 
-  dataRequest.dataRequestToken = null;
-  dataRequest.dataId = action.dataId;
-  const layerList = [...state.layerList];
-  return { ...state, layerList };
+  const layer = findLayerById(state, action.layerId);
+  const dataRequestIndex = layer.__dataRequests.indexOf(dataRequest);
+
+  const newDataRequests = [...layer.__dataRequests];
+  newDataRequests[dataRequestIndex] = { ...dataRequest };
+
+  const newDataRequest = newDataRequests[dataRequestIndex];
```

Comment:
Lines 462 through 464 are not needed. You can just set `dataRequestToken` and  `dataId` in line 460.